src/sound_len.rs

```diff
@@ -0,0 +1,157 @@
+use crate::error::{Error::SoundLen, Result};
+use std::{collections::HashMap, fs::File, path::Path, time::Duration};
+use symphonia::{
+    self,
+    core::{
+        codecs::DecoderOptions,
+        formats::FormatOptions,
+        io::MediaSourceStream,
+        meta::MetadataOptions,
+        probe::{Hint, ProbeResult},
+    },
+    default::{get_codecs, get_probe},
+};
+
+byond_fn!(fn sound_len(sound_path) {
+    match get_sound_length(sound_path) {
+        Ok(r) => return Some(r),
+        Err(e) => return Some(e.to_string())
+    }
+});
+
+fn get_sound_length(sound_path: &str) -> Result<String> {
+    let path = Path::new(sound_path);
+
+    // Gracefully exit if the filepath is invalid.
+    if !path.exists() {
+        return Err(SoundLen(format!("Bad path: {}", sound_path)));
+    }
+
+    // Try to open the file
+    let sound_src = match File::open(path) {
+        Ok(r) => r,
+        Err(_e) => return Err(SoundLen("Couldn't open file!".to_string())),
+    };
```

Comment:
Nit: It's good practice in general to just open it, and check if the error is file not found. 

https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.NotFound